stock_inventory/models/stock_inventory.py

```diff
@@ -0,0 +1,193 @@
+        active_rec = self.env["stock.inventory"].search([("state", "=", "in_progress")])
+        if active_rec:
+            raise ValidationError(
+                _("There's already an Adjustment in Process: %s") % active_rec.name
+            )
+        self.state = "in_progress"
+        if self.product_selection == "all":
+            for location in self._origin.location_ids:
+                self.stock_quant_ids = self.env["stock.quant"].search(
+                    [
+                        "|",
+                        ("location_id", "=", location.id),
+                        ("location_id", "in", location.child_ids.ids),
+                    ]
+                )
+        elif self.product_selection == "manual":
+            for location in self._origin.location_ids:
+                self.stock_quant_ids = self.env["stock.quant"].search(
+                    [
+                        ("product_id", "in", self.product_ids.ids),
+                        "|",
+                        ("location_id", "=", location.id),
+                        ("location_id", "in", location.child_ids.ids),
+                    ]
+                )
+        elif self.product_selection == "one":
+            for location in self._origin.location_ids:
+                self.stock_quant_ids = self.env["stock.quant"].search(
+                    [
+                        ("product_id", "in", self.product_ids.ids),
+                        "|",
+                        ("location_id", "=", location.id),
+                        ("location_id", "in", location.child_ids.ids),
+                    ]
+                )
+        elif self.product_selection == "lot":
+            for location in self._origin.location_ids:
+                self.stock_quant_ids = self.env["stock.quant"].search(
+                    [
+                        ("product_id", "in", self.product_ids.ids),
+                        ("lot_id", "in", self.lot_ids.ids),
+                        "|",
+                        ("location_id", "=", location.id),
+                        ("location_id", "in", location.child_ids.ids),
+                    ]
+                )
+        elif self.product_selection == "category":
+            for location in self._origin.location_ids:
+                self.stock_quant_ids = self.env["stock.quant"].search(
+                    [
+                        "|",
+                        ("product_id.categ_id", "=", self.category_id.id),
+                        ("product_id.categ_id", "in", self.category_id.child_id.ids),
+                        "|",
+                        ("location_id", "=", location.id),
+                        ("location_id", "in", location.child_ids.ids),
+                    ]
+                )
+        for quant in self.stock_quant_ids:
+            quant.to_do = True
+        return
+
+    def action_state_to_done(self):
+        self.state = "done"
+        for quant in self.stock_quant_ids:
+            quant.to_do = True
```

Comment:
It is still a manual action to confirm the quant count?

Reply:
Yes, confirming a count is still manual.

Reply:
Ok, this is very confusing? Should we at least give a warning to the user that the count(s) still needs to be confirmed? Setting a group to done for me means that the count is confirmed.

Reply:
What if a quant is being mutated which is currently in a stock adjustment which is not done yet, should we not block this? Same like not allowing 2 counts? Overall I have a sense the module is confusing now. Why do we add a state to stock_inventory if nothing is being done / blocked or checked? If it is simply a group of quants, then we can create a group without a workflow?